### PR TITLE
Multiple-diagrams: create a diagram file for each generator 

### PR DESCRIPTION
**Pull request by PerninGuillaume**
Before merging this branch, wide tests should be conducted to see if the division of diagram files is indeed faster
Signed-off-by: Guillaume Pernin <guillaume.pernin@rte-france.com>

### CODE REVIEW

File: sources/Outputs/src/Diagram.cpp [hunk 1]
```diff
@@ -28,30 +29,27 @@ Diagram::Diagram(DiagramDefinition&& def) : def_{std::forward<DiagramDefinition>
...
+
     writeTable(generator, buffer, Tables::TABLE_QMIN);
     writeTable(generator, buffer, Tables::TABLE_QMAX);
-  }
-
-  if (!empty) {
-    //Put buffer into file
-    std::ofstream ofs(def_.filename, std::ofstream::out);
+    boost::filesystem::path dir(def_.directoryPath);
+    std::string filename = dir.append(generator.id + outputs::constants::diagramFileSuffixExt).generic_string();
```
Comment (lecourtoisflo) [inline sources/Outputs/src/Diagram.cpp:51]: Use a function in constants to generate the filename of diagrams files

**Review by lecourtoisflo — CHANGES_REQUESTED** [review]

File: tests/outputs/TestDiagram.cpp [hunk 2]
```diff
@@ -8,22 +8,43 @@
...
 //
 
+#include "Constants.h"
 #include "Diagram.h"
 #include "Tests.h"
 
 #include <boost/filesystem.hpp>
 
+static void
+testMultiplesFilesEquality(const std::vector<dfl::algo::GeneratorDefinition>& generators, boost::filesystem::path outputDirectory, std::string basename,
```
Comment (rosiereflo) [inline tests/outputs/TestDiagram.cpp:18]: const & for everything

**Review by lecourtoisflo — APPROVED** [review]

File: tests/outputs/TestDiagram.cpp [hunk 3]
```diff
@@ -114,43 +130,37 @@ TEST(Diagram, writeWithCurveAndDefaultPoints) {
...
+  std::string prefixDir = "Empty";
+  boost::filesystem::path outputDirectory("results");
+  outputDirectory.append(basename);
 
-  if (!boost::filesystem::exists(outputPath)) {
-    boost::filesystem::create_directories(outputPath);
+  if (!boost::filesystem::exists(outputDirectory)) {
+    boost::filesystem::create_directories(outputDirectory);
   }
+  outputDirectory.append(prefixDir + dfl::outputs::constants::diagramDirectorySuffix);
```
Comment (rosiereflo) [inline tests/outputs/TestDiagram.cpp:150]: Please make sure the directory is empty

**Review by rosiereflo — CHANGES_REQUESTED** [review]

File: tests/outputs/TestDiagram.cpp (hunk 3 above)
Comment (PerninGuillaume) [reply tests/outputs/TestDiagram.cpp:150]: Do you mean I have to test twice the fact that it is empty ? It is already done 
` std::ifstream stream(directoryPath);
    ASSERT_TRUE(stream.fail()) << "The Diagram directory " << directoryPath << " has been created to even though there are no generators which model requires it."
                               << std::endl; ` 
here

File: sources/Outputs/src/Diagram.cpp [hunk 4]
```diff
@@ -28,30 +29,27 @@ Diagram::Diagram(DiagramDefinition&& def) : def_{std::forward<DiagramDefinition>
...
+    std::ofstream ofs(filename, std::ofstream::out);
```
[the remark this answers is not in the document]
Comment (PerninGuillaume) [reply sources/Outputs/src/Diagram.cpp:52]: Did you mean by "the diagram is empty" the case where it has no Reactive curve points ?

File: tests/outputs/TestDiagram.cpp (hunk 3 above)
Comment (rosiereflo) [reply tests/outputs/TestDiagram.cpp:150]: ok!

File: sources/Outputs/src/Diagram.cpp (hunk 4 above)
[the remark this answers is not in the document]
Comment (rosiereflo) [reply sources/Outputs/src/Diagram.cpp:52]: yes, but I'm not sure of this one

[the remark this answers is not in the document]
Comment (PerninGuillaume) [reply sources/Outputs/src/Diagram.cpp:52]: When we have no Reactive curve points in a diagram we dump it with the default values of pmin,pmax,qmin and qmax

**Review by rosiereflo — APPROVED** [review]

**Review by marcochiaramello — APPROVED** [review]

**Pull request merged**